pkg/apis/extensions/types.go

```diff
@@ -278,6 +283,40 @@ type DeploymentStatus struct {
 
 	// Total number of unavailable pods targeted by this deployment.
 	UnavailableReplicas int32 `json:"unavailableReplicas,omitempty"`
+
+	// Represents the latest available observations of a deployment's current state.
+	Conditions []DeploymentCondition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+}
+
+type DeploymentConditionType string
+
+// These are valid conditions of a deployment.
+const (
+	// DeploymentAvailable means the deployment is available, ie. at least the minimum available
+	// replicas required are up and running for at least minReadySeconds.
+	DeploymentAvailable DeploymentConditionType = "Available"
+	// DeploymentProgressing means the deployment is progressing. Progress for a deployment is
+	// considered when a new replica set is created or adopted, and when new pods scale up or
+	// old pods scale down. Progress is not estimated for paused deployments, when the deployment
+	// needs to rollback, or when progressDeadlineSeconds is not specified.
+	DeploymentProgressing DeploymentConditionType = "Progressing"
+	// DeploymentReplicaFailure is added in a deployment when one of its pods
+	// fails to be created or deleted.
+	DeploymentReplicaFailure DeploymentConditionType = "ReplicaFailure"
```

Comment:
Shouldn't there be a state which just means the Deployment has been marked failed, or is ReplicaFailure that state ? Why is not called `Failure' rather than qualifying it as ReplicaFailure. ?